storage/src/rocksdb_client.cpp

```diff
@@ -81,35 +92,24 @@ bool Client::isNew() {
  * Uses the RocksDBClient object variables m_dbPath and m_dbInstance to
  * establish a connection with RocksDB by creating a RocksDB object.
  *
- *  @return GeneralError in case of error in connection, else OK.
+ *  @throw GeneralError in case of error in connection, else OK.
  */
-Status Client::init(bool readOnly) {
-  ::rocksdb::DB *db;
+void Client::init(bool readOnly) {
+  ::rocksdb::DB* db;
   ::rocksdb::Options options;
+  ::rocksdb::TransactionDBOptions txn_options;
   options.create_if_missing = true;
   options.comparator = m_comparator;
-
   ::rocksdb::Status s;
   if (readOnly) {
     s = ::rocksdb::DB::OpenForReadOnly(options, m_dbPath, &db);
+    m_dbInstance.reset(db);
   } else {
-    s = ::rocksdb::DB::Open(options, m_dbPath, &db);
+    s = ::rocksdb::TransactionDB::Open(options, txn_options, m_dbPath, &txn_db_);
+    m_dbInstance.reset(txn_db_->GetBaseDB());
   }
-  m_dbInstance.reset(db);
-
-  if (!s.ok()) {
-    LOG_ERROR(logger, "Failed to open rocksdb database at "
-                                << m_dbPath << " due to " << s.ToString());
-    return Status::GeneralError("Database open error");
-  }
-
-  g_rocksdb_called_read = 0;
-
-  // TODO(Shelly): Update for measurements. Remove when done as well as other
-  // g_rocksdb_* variables.
-  g_rocksdb_print_measurements = false;
```

Comment:
Why removed?

Reply:
it's moved to the beginning of the file where the globals are are declared